src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -242,61 +287,92 @@ impl Time {
         }
     }
 
+    fn split_datetime(s: &str) -> (Vec<&str>, &str) {
+        let trimmed = s.trim();
+        if trimmed.is_empty() {
+            return (vec![], "");
+        }
+        let (parts, fracs) = if let Some(i) = trimmed.rfind('.') {
+            (&trimmed[..i], &trimmed[i + 1..])
+        } else {
+            (trimmed, "")
+        };
+        (Time::parse_datetime_format(parts), fracs)
+    }
+
     pub fn parse_utc_datetime(s: &str, fsp: i8) -> Result<Time> {
         Time::parse_datetime(s, fsp, Tz::utc())
     }
 
+    pub fn parse_utc_datetime_from_float_string(s: &str, fsp: i8) -> Result<Time> {
+        Time::parse_datetime_from_float_string(s, fsp, Tz::utc())
+    }
+
     pub fn parse_datetime(s: &str, fsp: i8, tz: Tz) -> Result<Time> {
+        Time::parse_datetime_internal(s, fsp, tz, false)
+    }
+
+    pub fn parse_datetime_from_float_string(s: &str, fsp: i8, tz: Tz) -> Result<Time> {
+        Time::parse_datetime_internal(s, fsp, tz, true)
+    }
+
+    fn parse_datetime_internal(s: &str, fsp: i8, tz: Tz, is_float: bool) -> Result<Time> {
```

Comment:
Why not make it pub and remove `parse_datetime` and `parse_datetime_from_float_string`? 
`parse_utc_datetime` and `parse_utc_datetime_from_float_string` can also be merge into one func.

Reply:
@solotzg  `parse_datetime_from_float_string` is target for only few scenes. In most cases, the user don't care the difference on behavior. So I think it's better to make it transparent in these scenarios. 
If we merge `parse_utc_datetime` and `parse_utc_datetime_from_float_string` into one function, the extra parameter may people a little confused

Reply:
@breeswish

Reply:
I think we should minimize the difference between TiKV's coprocessor implementation and TiDB's (however we need to preserve some styles from Rust), so that other people won't put much effort on finding corresponding implementation.

Reply:
@breeswish yes, TiDB also seperates `parse_utc_datetime_from_float_string`  from  `parse_utc_datetime `